src/core/models/outputs/output_extended.c

```diff
@@ -1,4 +1,300 @@
+          break;
+        case FEAT_METADATA_BLOCK: {
+          feat_metadata_blk_t* metadata = (feat_metadata_blk_t*)feat->blk->block;
+          res = feat_blk_list_add_metadata(&output->feature_blocks, metadata->data, metadata->data_len);
+          break;
+        }
+        case FEAT_INDEXATION_BLOCK: {
+          feat_indexaction_blk_t* indexation = (feat_indexaction_blk_t*)feat->blk->block;
+          res = feat_blk_list_add_indexaction(&output->feature_blocks, indexation->tag, indexation->tag_len);
+          break;
+        }
+      }
+      if (res == -1) {
+        printf("[%s:%d] can not add feature block to extended output\n", __func__, __LINE__);
+        output_extended_free(output);
+        return NULL;
+      }
+    }
+  }
+
+  return output;
+}
+
+void output_extended_free(output_extended_t* output) {
+  if (output) {
+    if (output->address) {
+      free_address(output->address);
+    }
+    if (output->native_tokens) {
+      native_tokens_free(&output->native_tokens);
+    }
+    if (output->feature_blocks) {
+      free_feat_blk_list(output->feature_blocks);
+    }
+    free(output);
+  }
+}
+
+size_t output_extended_serialize_len(output_extended_t* output) {
+  if (output == NULL) {
+    printf("[%s:%d] invalid parameters\n", __func__, __LINE__);
+    return 0;
+  }
+
+  size_t length = 0;
+
+  // output type
+  length += sizeof(uint8_t);
+  // address
+  length += address_serialized_len(output->address);
+  // amount
+  length += sizeof(uint64_t);
+  // native tokens
+  length += native_tokens_serialize_len(&output->native_tokens);
+  // feature blocks
+  length += feat_blk_list_serialize_len(output->feature_blocks);
+
+  return length;
+}
+
+size_t output_extended_serialize(output_extended_t* output, byte_t buf[], size_t buf_len) {
+  if (output == NULL || buf == NULL || buf_len == 0) {
+    printf("[%s:%d] invalid parameters\n", __func__, __LINE__);
+    return 0;
+  }
+
+  size_t expected_bytes = output_extended_serialize_len(output);
+  if (buf_len < expected_bytes) {
+    printf("[%s:%d] buffer size is insufficient\n", __func__, __LINE__);
+    return 0;
+  }
+
+  byte_t* offset = buf;
+
+  // fill-in output type, set to value 3 to denote a Extended Output
+  memset(offset, 3, sizeof(uint8_t));
+  offset += sizeof(uint8_t);
+
+  // address
+  offset += address_serialize(output->address, offset, address_serialized_len(output->address));
+
+  // amount
+  memcpy(offset, &output->amount, sizeof(uint64_t));
+  offset += sizeof(uint64_t);
+
+  // native tokens
+  if (output->native_tokens) {
+    offset +=
+        native_tokens_serialize(&output->native_tokens, offset, native_tokens_serialize_len(&output->native_tokens));
+  } else {
+    memset(offset, 0, sizeof(uint16_t));
+    offset += sizeof(uint16_t);
+  }
+
+  // feature blocks
+  if (output->feature_blocks) {
+    offset +=
+        feat_blk_list_serialize(output->feature_blocks, offset, feat_blk_list_serialize_len(output->feature_blocks));
```

Comment:
the serialized data is valid if and only if block types are in ascending order. file an issue to fix it, #224